amqp/connection.py

```diff
@@ -754,7 +754,7 @@ def heartbeat_tick(self, rate=2):
         self.prev_sent, self.prev_recv = sent_now, recv_now
 
         # send a heartbeat if it's time to do so
-        if now > self.last_heartbeat_sent + self.heartbeat:
+        if now > self.last_heartbeat_sent + self.heartbeat / rate:
```

Comment:
we need to verify this change with unit test

Reply:
Also, I need at least to check the divide by 0 and remove the comment about deprecated parameter

Reply:
done